src/lib.rs

```diff
@@ -90,6 +90,17 @@ pub trait Group:
     /// Doubles this element.
     #[must_use]
     fn double(&self) -> Self;
+
+    /// Perform multi-scalar multiplication of the form `sum(base[i] * scalar[i])`.
+    fn multiscalar_mul(bases: &[Self], scalars: &[Self::Scalar]) -> Self {
```

Comment:
Since you're zipping them anyway below, perhaps it would make sense to take something like an `impl IntoIterator<Item = (&Self, &Self::Scalar)>`?

Reply:
It could be for this default implementation, but I think that would be too limiting in general. It's usually beneficial to have random access to the elements.